starknet_py/devnet_utils/devnet_client.py

```diff
@@ -0,0 +1,376 @@
+                "unit": unit.upper() if isinstance(unit, str) else unit.value,
+            },
+        )
+
+        return cast(MintResponse, MintResponseSchema().load(res))
+
+    async def get_account_balance(
+        self,
+        address: Hash,
+        unit: Union[PriceUnit, str] = PriceUnit.WEI,
+        block_tag: str = "latest",
+    ) -> BalanceRecord:
+        """
+        Get the balance of the given account.
+
+        :param address: Address of the account contract.
+        :param unit: Literals `"FRI"` or `"WEI"` defaults to `"WEI"`.
+        :param block_tag: Literals `"pending"` or `"latest"`, defaults to `"latest"`.
+        """
+
+        res = await self._devnet_client.call(
+            method_name="getAccountBalance",
+            params={
+                "address": _to_rpc_felt(address),
+                "unit": unit.upper() if isinstance(unit, str) else unit.value,
+                "block_tag": block_tag,
+            },
+        )
+
+        return cast(BalanceRecord, BalanceRecordSchema().load(res))
+
+    async def create_block(self) -> str:
+        """
+        Create a new block.
+        """
+
+        res = await self._devnet_client.call(method_name="createBlock")
+
+        return res["block_hash"]
+
+    async def abort_block(self, starting_block_hash: Hash) -> List[str]:
+        """
+        This functionality allows simulating block abortion that can occur on mainnet.
+        It is supported in the `--state-archive-capacity full` mode.
+
+        :param starting_block_hash: The state of Devnet will be reverted to the state before `starting_block_hash`.
+        """
+
+        res = await self._devnet_client.call(
+            method_name="abortBlocks",
+            params={"starting_block_hash": _to_rpc_felt(starting_block_hash)},
+        )
+
+        return res["aborted"]
+
+    async def dump(self, path: str):
+        """
+        Dump the state of the devnet to a file.
+        Dumping on request requires providing `--dump-on` mode on the startup.
+
+        :param path: Path to the file.
+        """
+
+        await self._devnet_client.call(
+            method_name="dump",
+            params={"path": path},
+        )
+
+    async def load(self, path: str):
+        """
+        Load the state of the devnet from a file.
+
+        :param path: Path to the file.
+        """
+
+        await self._devnet_client.call(
+            method_name="load",
+            params={"path": path},
+        )
+
+    async def restart(self):
+        """
+        Restart the devnet.
+        """
+
+        await self._devnet_client.call(
+            method_name="restart",
+        )
+
+    async def postman_load(
+        self, network_url: str, address: Optional[str] = None
+    ) -> str:
+        """
+        Loads a `MockStarknetMessaging
+        <https://github.com/0xSpaceShard/starknet-devnet-rs/blob/main/contracts/l1-l2-messaging/solidity/src/MockStarknetMessaging.sol>`_
```

Comment:
It is usually better to include a commit in the github link, as this will ensure the link doesn't break in case the file is moved.